sycl/source/detail/platform_impl.cpp

```diff
@@ -20,6 +20,20 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 
+static bool IsBannedPlatform(platform Platform) {
+  auto IsNVIDIAOpenCL = [](platform Platform) {
+    if (Platform.is_host())
+      return false;
+
+    const bool HasCUDA = Platform.get_info<info::platform::name>().find(
+                             "NVIDIA CUDA") != std::string::npos;
+    const auto Backend =
+        detail::getSyclObjImpl(Platform)->getPlugin().getBackend();
+    return (HasCUDA && Backend == backend::opencl);
+  };
+  return IsNVIDIAOpenCL(Platform);
```

Comment:
The intention is to disable NVIDIA OpenCL platform for the foreseeable future, among many reasons, because its not really needed when having the CUDA backend. See https://github.com/intel/llvm/issues/1665 for a longer discussion about this.

Reply:
Thanks for pointing to the discussion. Should we at least check that PI CUDA backend is available before shooting the OpenCL CUDA backend?

Reply:
So you mean, if the DPCPP is not built with CUDA support, the NVIDIA OpenCL should still be available for device selection? That is still untested. 
Maybe its better to have an env flag to disable the banned platform list and let users shoot themselves in the foot if they want. But I think that should happen on a separate PR.

Reply:
Can't we use existing whitelist functionality to filter/ban this? I am OK with doing it separately.